Bottom: b09ed2b8067610b5c4972e32ab4ed003408c2c4d
Top:    b99badef4f876bc7e1fc9eefab867b30f0916e72
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-30 19:08:26 +0100

Refresh of convert-board-objects-to-glist

---

diff --git a/src/puller.c b/src/puller.c
index 2238204..488748d 100644
--- a/src/puller.c
+++ b/src/puller.c
@@ -2651,21 +2651,19 @@ GlobalPuller(int argc, char **argv, int x, int y)
   printf("\nlines\n");
 #endif
 
-  /* We do this backwards so we don't have to edit the extras.  */
-  for (i=CURRENT->LineN-1; i>=0; i--)
+  LINE_LOOP (CURRENT);
     {
-      LineTypePtr line = & CURRENT->Line[i];
       if (LINE2EXTRA (line)->deleted)
 	RemoveLine (CURRENT, line);
     }
+  END_LOOP;
 
-  /* We do this backwards so we don't have to edit the extras.  */
-  for (i=CURRENT->ArcN-1; i>=0; i--)
+  ARC_LOOP (CURRENT);
     {
-      ArcTypePtr arc = & CURRENT->Arc[i];
       if (ARC2EXTRA (arc)->deleted)
 	RemoveArc (CURRENT, arc);
     }
+  END_LOOP;
 
   g_hash_table_unref (lines);
   g_hash_table_unref (arcs);
